core/src/main/java/tc/oc/pgm/scoreboard/SidebarMatchModule.java

```diff
@@ -258,9 +260,55 @@ private Component renderTitle(final Config config, final MapInfo map) {
       return header.colorIfAbsent(NamedTextColor.AQUA);
     }
 
-    final Component game = map.getGamemode();
-    if (game != null) {
-      return game.colorIfAbsent(NamedTextColor.AQUA);
+    final Component gamemode = map.getGamemode();
+    if (gamemode != null) {
+      return gamemode.colorIfAbsent(NamedTextColor.AQUA);
+    }
+
+    final Collection<Gamemode> gamemodes = map.getGamemodes();
+    if (!gamemodes.isEmpty()) {
+      Component gamemodeTitle = null;
+      List<Component> gamemodeComponents = new ArrayList<>();
+      for (Gamemode value : gamemodes) {
+        Component gmComponent;
+        // Make this smarter using MAX_LENGTH
+        if (gamemodes.size() <= 1) {
+          gmComponent = Component.translatable("gamemode." + value.getId() + ".name");
+        } else {
+          gmComponent = Component.translatable("gamemode." + value.getId() + ".acronym");
+        }
+        gamemodeComponents.add(gmComponent);
+      }
+      switch (gamemodes.size()) {
+        case 1:
+          gamemodeTitle = gamemodeComponents.get(0).colorIfAbsent(NamedTextColor.AQUA);
+          break;
+        case 2:
+          gamemodeTitle =
+              Component.translatable(
+                      "misc.list.pair", gamemodeComponents.get(0), gamemodeComponents.get(1))
+                  .colorIfAbsent(NamedTextColor.AQUA);
+          break;
+        case 3:
+          gamemodeTitle =
+              Component.translatable(
+                      "misc.list.start", gamemodeComponents.get(0), gamemodeComponents.get(1))
+                  .append(
+                      Component.translatable(
+                          "misc.list.end", TextColor.fromHexString(""), gamemodeComponents.get(2)))
+                  .colorIfAbsent(NamedTextColor.AQUA);
+          break;
+        case 4:
+          gamemodeTitle =
+              Component.translatable(
+                      "misc.list.start", gamemodeComponents.get(0), gamemodeComponents.get(1))
+                  .append(
+                      Component.translatable(
+                          "misc.list.end", gamemodeComponents.get(2), gamemodeComponents.get(3)))
+                  .colorIfAbsent(NamedTextColor.AQUA);
+          break;
+      }
```

Comment:
I'm 100% sure this has to be a util somewhere, if it really isn't, make it one.

Also, what about over 4? i know it's not going to be common but you still have to support N items on the list